### PR TITLE
azure_rm_virtualmachine - adding support for StandardSSD_LRS

### PR DESCRIPTION
**Pull request by rnsc**
##### SUMMARY

StandardSSD_LRS have been introduced (generally available) in June 2018.
This change will add support for this.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

azure_rm_virtualmachine
##### ANSIBLE VERSION

```paste below
ansible 2.6.4
  config file = /home/rnsc/<redacted>/ansible/ansible.cfg
  configured module search path = [u'/home/rnsc/VE/local/lib/python2.7/site-packages/ara/plugins/modules']
  ansible python module location = /home/rnsc/VE/local/lib/python2.7/site-packages/ansible
  executable location = /home/rnsc/VE/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]
```

##### ADDITIONAL INFORMATION

This is my first PR, I hope I understood the procedure correctly.
Not sure how to work it out if I want to have the feature backported to 2.6 along with 2.7 support.

```paste below

```

### CODE REVIEW

**Review by yuwzho — APPROVED** [review]

**Comment by yuwzho** [comment]
@Fred-sun Please test this pr by creating a vm with `StandardSSD_LRS`

**Comment by rnsc** [comment]
This is the reference ticket:
https://github.com/ansible/ansible/issues/46537

**Comment by rnsc** [comment]
Hello @yuwzho 
I think I overlooked this document:
https://docs.microsoft.com/en-us/azure/virtual-machines/windows/faq-for-disks#standard-ssd-disks
The Compute API version needs to be updated to at least version _2018-04-01_ :

https://github.com/ansible/ansible/blob/23c2dc3b2a9066f1604c0c2bab13f1cacefb0864/lib/ansible/module_utils/azure_rm_common.py#L55-L63

**Comment by Fred-sun** [comment]
@yuwzho  I have tested this PR , but when I try to create VM with StandardSSD_LRS, Create fail. my test YML and error message as below. Thanks!
```e.g.
    - name: "azure_rm_virtualmachine"
      azure_rm_virtualmachine:
        resource_group: "{{ resource_group }}"
        name: "{{ vm_name }}"
        managed_disk_type: StandardSSD_LRS
        #managed_disk_type: Standard_LRS
        os_disk_caching: ReadWrite
        network_interfaces: "{{ test_nic }}"
        virtual_network_name: "{{ virtualnetwork }}"
        subnet_name: "{{ subnet }}"
        vm_size: "Standard_D4"
        admin_username: azureuser
        admin_password: Password@0329
        image:
          offer: UbuntuServer
          publisher: Canonical
          sku: 16.04-LTS
          version: latest
```
```e.g.
TASK [azure_rm_virtualmachine] *************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Error creating or updating virtual machine myVM01 - 'StandardSSD_LRS' is not valid value for enum <enum 'StorageAccountTypes'>"}
	to retry, use: --limit @/home/fred/task/46537/46537.retry

PLAY RECAP *********************************************************************
localhost                  : ok=7    changed=6    unreachable=0    failed=1

**Comment by Fred-sun** [comment]
@yuwzho @rnsc Could you help take a look this PR? Create VM with StandardSSD_LRS got fail. Thanks!

**Comment by yuwzho** [comment]
@Fred-sun Have you ever installed rnsc's branch?

**Comment by yuwzho** [comment]
@rnsc Please go ahead to update the common file. The CI should detect whether there is  anything breaking if the API version updated. Also note, the SDK library version is controlled by file

https://github.com/ansible/ansible/blob/6a1aa51a743e9bdfdf2b5ed24e36aa142e5dc4bc/packaging/requirements/requirements-azure.txt#L8 

and 

https://github.com/ansible/ansible/blob/6a1aa51a743e9bdfdf2b5ed24e36aa142e5dc4bc/test/runner/requirements/integration.cloud.azure.txt#L8

**Comment by Fred-sun** [comment]
@yuwzho Yes, I have installed rnsc's branch, if not install new. it will get error as below. please review, Thanks!
```e.g.
TASK [azure_rm_virtualmachine] *************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "value of managed_disk_type must be one of: Standard_LRS, Premium_LRS, got: StandardSSD_LRS"}
	to retry, use: --limit @/home/fred/task/46537/46537.retry

PLAY RECAP *********************************************************************
localhost                  : ok=9    changed=0    unreachable=0    failed=1

**Comment by Fred-sun** [comment]
@rnsc Thanks for the contribution, Could you help change according by the comment's? Thanks!

**Comment by Fred-sun** [comment]
@rnsc thanks for the contribution, Are you still here? Could you help recheck this PR? Thanks!

**Comment by rnsc** [comment]
@Fred-sun hey, I haven't had enough time recently to check this out again...

**Comment by Fred-sun** [comment]
@rnsc Thanks for your reply and contribution

**Comment by Fred-sun** [comment]
@yuwzho An error was encountered while checking the StorageAccountType , The StandardSSD_LRS variable should be added to the compute_management_client_enums module too. I will continue to pay attention to this issue. Thanks!

**Comment by Fred-sun** [comment]
@rnsc Thanks for your contribution, Are you still pay attention on this PR? if need help, you can write the demand into comment, the maintainer will give you help. Thanks!

**Comment by Fred-sun** [comment]
kindly ping

**Comment by Fred-sun** [comment]
@yuwzho  how about the PR going ahead? Could you help re-check and push for this merge ?

**Comment by Fred-sun** [comment]
@yuwzho

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
rebuild_merge

**Comment by gundalow** [comment]
@rnsc Thank you for this PR.
This has been merged into `devel` for release in Ansible 2.8

We don't backport features to older releases.

**Comment by mohag** [comment]
`UltraSSD_LRS` should probably also be supported

**Comment by yungezz** [comment]
easy_fix

**Comment by yungezz** [comment]
@mohag  thanks for sharing. Seems UltraSSD_LRS in preview with lots of limitation. https://docs.microsoft.com/en-us/azure/virtual-machines/windows/disks-ultra-ssd . we can add it when majority of users could use the feature.

**Comment by mohag** [comment]
@yungezz : I'm aware of that, if the API rejects it in unsupported places with a reasonable message that gets passed on to the Ansible user, I would prefer that to waiting for the next Ansible release after it goes GA before it is supported. (I currently need to use another tool to deploy my VMs (or manually patch the module) due to `StandardSSD_LRS` not working in the stable version)

**Pull request merged**
